ml_genn/ml_genn/compilers/event_prop_compiler.py

```diff
@@ -1240,16 +1243,18 @@ def _build_adjoint_system(self, model: AutoNeuronModel,
         # generate adjoint ODE
         # assume that neuron variables do not appear in rhs of post-synapse ODEs
         # therefore, we can do these independent of synapse equations
-        saved_vars = set()
+        saved_vars_timestep = set()
+        saved_vars_spike = set()
         dl_dt = {}
         for sym in model.dx_dt.keys():
             o = sum(sympy.diff(expr2, sym) * _get_lmd_sym(sym2)
                     for sym2, expr2 in model.dx_dt.items())
             
             # collect variables they might need to go into a ring buffer:
-            # **TODO** helper
-            saved_vars.update({sym2.name for sym2 in model.dx_dt.keys()
-                               if o.has(sym2)})
+            # JAMIE - I feel uneasy for this to go through the solvers but it works
+            # as the decorated $xxx vars are not the dynamic vars of this backward system
```

Comment:
```suggestion
```
Yes, so did I but it seems sympy seems to happily allows variable names that C definitely doesn't 😄